changelog/@unreleased/pr-4387.v2.yml

```diff
@@ -0,0 +1,5 @@
+type: improvement
+improvement:
+  description: Improved resource management of CQL resources. Custom management logic should implement `CassandraAsyncKeyValueServiceFactory`.
```

Comment:
I'd word the 2nd sentence slightly differently: Users desiring custom management logic should implement `CassandraAsyncKeyValueServiceFactory`.

Reply:
Rewritten.